numba/np/ufunc/_internal.c

```diff
@@ -545,16 +544,6 @@ dufunc__add_loop(PyDUFuncObject * self, PyObject * args)
                                         arg_types_arr, data_ptr) < 0) {
             goto _dufunc__add_loop_fail;
         }
-    } else if (PyUFunc_ReplaceLoopBySignature(ufunc,
-                                              (PyUFuncGenericFunction)loop_ptr,
-                                              arg_types_arr, &old_func) == 0) {
-        /* TODO: Consider freeing any memory held by the old loop (somehow) */
-        for (idx = 0; idx < ufunc->ntypes; idx++) {
-            if (ufunc->functions[idx] == (PyUFuncGenericFunction)loop_ptr) {
-                ufunc->data[idx] = data_ptr;
-                break;
-            }
-        }
```

Comment:
I found a case that reaches this:

```python
from numba import vectorize, types


@vectorize
def foo(x):
    return x

foo.add(tuple([types.intp]))
foo.add(tuple([types.intp]))
```

I think it is valid for Numba to reject the second `.add`.